File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    N = len(citations); citations.sort(reverse=True)
    
    h_idx = max(citations)
    while h_idx >= 0:
        l, s = 0, 0
        for c in citations:
            if c >= h_idx: l += 1
            else: s += 1
        if l >= h_idx and s <= h_idx: return h_idx
        else: h_idx -= 1

File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<int> citations) {
    
    sort(citations.begin(), citations.end(), [](const int &a, const int &b)
         {
             return a > b;
         });
    
    // for (auto &it: citations)
    // {
    //     cout << it << " ";
    // }
    
    int H_idx = citations[0];
    while (H_idx >= 0)
    {
        int larger_cnt = 0;
        for (auto &it : citations)
        {
            if (H_idx <= it)
            {
                larger_cnt++;
            }
        }
        if (larger_cnt >= H_idx)
        {
            return H_idx;
        }
        else
        {
            H_idx--;
        }
    }
}